### PR TITLE
Validate club website, prepend http://.

### LINKED ISSUE #424: URL validation on 'website' field in club modal should be friendlier
As mentioned in https://github.com/mozilla/teach.webmaker.org/issues/352#issuecomment-84214121, we are currently relying on `type="url"` to validate the website field in Club modals. This ultimately means that the validation is at the discretion of the host browser's built-in HTML5 form validation; on some browsers it says "Please enter a URL" while on others it may say "Please enter a web address that starts with http:// or https://" while others may say something different.

This isn't super ideal. Also, if the user enters "example.org" we could always prepend "http://" before sending the data to the server (the server uses a [`URLField`](https://docs.djangoproject.com/en/1.7/ref/models/fields/#urlfield) and thus expects any non-blank value to be a http or https URL).

### PR DESCRIPTION
**Pull request by toolness**
This fixes #424.

### CODE REVIEW

File: pages/clubs.jsx [hunk 1]
```diff
@@ -231,6 +241,9 @@ var validateClub = function(clubState) {
   if (!clubState.location) {
     errors.push("You must provide a location for your club.");
   }
+  if (clubState.website && !/\./.test(clubState.website)) {
```
Comment (toolness) [inline pages/clubs.jsx]: Note that this is super basic validation; we're essentially just testing to make sure that a `.` appears somewhere in what they've typed. It's way less rigorous than [what Django uses](http://stackoverflow.com/a/7160778), which means that technically, users could submit a URL that our front-end accepts but the teach-api backend doesn't (in which case the user will just get a cryptic message saying that something went wrong and  to retry later).

File: test/browser/clubs-page.test.jsx [hunk 2]
```diff
@@ -283,7 +309,7 @@ describe("ClubsPage.ModalAddOrChangeYourClub", function() {
           validationErrors: ['old error that should be removed'],
           name: 'blorpy',
           location: 'chicago',
-          website: 'http://example.org',
+          website: 'example.org',
```
Comment (toolness) [inline test/browser/clubs-page.test.jsx]: This has the side effect of ensuring that submitting the form normalizes the `website` field into a proper URL.

**Comment by mmmavis** [comment]
r+ :ok_hand:

**Comment by mmmavis** [comment]
I'm wondering if we should add some visual hints to required fields. (asterisks? maybe?)

**Comment by toolness** [comment]
Thanks!

> I'm wondering if we should add some visual hints to required fields. (asterisks? maybe?)

Yeah, that was #577, but it does need some design work as I mention in https://github.com/mozilla/teach.webmaker.org/issues/577#issuecomment-91236434... and then Hannah closed it...

**Comment by mmmavis** [comment]
Let me open a new ticket and reference to [your comment](https://github.com/mozilla/teach.webmaker.org/issues/577#issuecomment-91236434)

**Comment by mmmavis** [comment]
k, just entered a new ticket for that: https://github.com/mozilla/teach.webmaker.org/issues/744

**Pull request merged**